plenum/server/primary_selector.py

```diff
@@ -209,10 +222,10 @@ def has_view_change_from_primary(self) -> bool:
                         self.viewNo, self.node.totalNodes))
                 return False
             else:
-                self._has_view_change_from_primary = True
+                logger.debug('{} received ViewChangeDone from primary {}'
+                             .format(self, next_primary_name))
 
-        logger.debug('{} received ViewChangeDone from primary {}'
-                     .format(self, self.next_primary_node_name(0)))
+        self._has_view_change_from_primary = True
```

Comment:
Why it was moved from else? We should set it to True only if next_primary_name is in self._view_change_done, shouldn't we?

Reply:
refactored